src/Microsoft.Graph.Core/Requests/Middleware/RetryHandler.cs

```diff
@@ -58,7 +58,7 @@ protected async override Task<HttpResponseMessage> SendAsync(HttpRequestMessage
 
             var response = await base.SendAsync(httpRequest, cancellationToken);
 
-            if(ShouldRetry(response) && httpRequest.IsBuffered() && RetryOption.MaxRetry > 0 && RetryOption.ShouldRetry(RetryOption.Delay, 0, response))
+            if(ShouldRetry(response) && httpRequest.IsBuffered() && RetryOption.RetriesTimeLimit > 0 && RetryOption.ShouldRetry(RetryOption.Delay, 0, response))
```

Comment:
This may change behavior, although I'm not certain yet that this is a negative thing since this is retry, and I can't off the top of my head figure a way that a customer could peg to the behavior. 

We should default to use MaxRetry and only use RetriesTimeLimit if the customer specifies it. I'd like to hear folks thoughts on this. This will affect the review on parts of this PR.

Also, please add comments to describe what this long condition is doing, for our future us.